From: James Smart <james.smart@broadcom.com>
Date: Wed, 7 Mar 2018 09:20:07 +0100
Subject: [PATCH] nvme-fc: kick admin requeue list on disconnect
References: bsc#1077241
Patch-Mainline: Never, SLE12-SP3 specific

We need to kick the admin requeue list on disconnect, too, otherwise
any outstanding commands will never terminated and the disconnect
hangs.

Signed-off-by: James Smart <james.smart@broadcom.com>
Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/fc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/nvme/host/fc.c b/drivers/nvme/host/fc.c
index d46942828a34..2ac55e6520f2 100644
--- a/drivers/nvme/host/fc.c
+++ b/drivers/nvme/host/fc.c
@@ -2882,6 +2882,7 @@ nvme_fc_delete_association(struct nvme_fc_ctrl *ctrl)
 
 	/* re-enable the admin_q so anything new can fast fail */
 	blk_mq_start_stopped_hw_queues(ctrl->ctrl.admin_q, true);
+	blk_mq_kick_requeue_list(ctrl->ctrl.admin_q);
 
 	nvme_fc_ctlr_inactive_on_rport(ctrl);
 }
-- 
2.12.3

